### PR TITLE
Budged App - Full site

### PR DESCRIPTION
**Pull request by angeluray**
## Hi good reviewer, I am glad to see you again 🖖🏼 🤖 

Happy Greeting

## Basic update check ✔️ 

- Created a new branch according to GitHub flow rules.
- Added linters and make sure they work correctly.
- Added specs for main functionalities.
- Added Splash screen, Categories, Transactions(For each category) and forms sections to add categories and transactions, besides Devise ones.
- Removed N+1 issues.
- Added styles following the design.
- Deployed.

## Visual check ✔️ 

Specs screenshot:

image_2022-12-21_19-24-54

**Note:**  *If you have further questions about the project or you see something that could be fixed without asking for a new review just let me know or mark it as optional changes, I'll 100% consider your suggestions and I'll be completely willing to fix it 😃 .*

### CODE REVIEW

File: app/controllers/relations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class RelationsController < ApplicationController
+  def index
+    @group = Group.find(params[:group_id])
+    @charges = @group.relations.includes(:charge).map(&:charge)
+  end
```
Comment (TedLivist) [inline app/controllers/relations_controller.rb]: - [ ] The expenses should be ordered by the most recent. Kindly implement that. If I have two orders, the first one should be the one I added last the last one should be the one I added first.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      <div class="new-my-form">
+        <%= image_tag("https://img.icons8.com/ios-glyphs/30/FFFFFF/menu--v1.png") %>
+        <h2>CATEGORIES</h2>
+        <%if user_signed_in? %>
+        <%= button_to "Sign Out", destroy_user_session_path, class: "signout", method: :delete %>
+        <% end %>
+      </div>
+      <% else %>
+      <div class="new-my-form">
+        <%= link_to '<', root_path %>
```
Comment (TedLivist) [inline app/views/layouts/application.html.erb:33]: - [ ] This "<" button should work differently for different pages. When I am on the page for adding an expense (transaction) and I click this button I should be taken to the transaction page for that category. Kindly implement that as well.

File: app/views/charges/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  </div>
+
+  <div class="add-t-container">
+    <%= form.submit 'SAVE', class: "add-transaction" %>
+  </div>
+
+  <div class="button-back">
+  <%= link_to "Back to charges", charges_path, class: "back-to" %>
+  </div>
+<% end %>
```
Comment (TedLivist) [inline app/views/charges/_form.html.erb]: - [ ] Nice job handling errors for the name and amount fields, I would also recommend you handle errors for the category field. When I try to add a transaction without a category I get this page.

  I think we can handle this better for a better experience by making the field compulsory and rendering error message just like the ones for name and amount.

  image

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @angeluray,

Good job so far!
✔️ Nice design
✔️ Implementation of validations
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/relations_controller.rb (hunk 1 above)
Comment (angeluray) [reply app/controllers/relations_controller.rb]: @TedLivist Hi, done 👍🏼 .

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (angeluray) [reply app/views/layouts/application.html.erb:33]: Done 👍🏼  @TedLivist

File: app/views/charges/_form.html.erb (hunk 3 above)
Comment (angeluray) [reply app/views/charges/_form.html.erb]: Thank you @TedLivist for your feeedback. I already implemented all the changes requested 👍🏼 .

**Review by prolajumokeoni — APPROVED** [review]
Hi @angeluray 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀  
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**